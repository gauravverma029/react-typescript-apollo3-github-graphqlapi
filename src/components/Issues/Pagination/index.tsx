import React from "react";

type Props = {
  onClickLoadMore: () => void;
};

const Pagination: React.FC<Props> = ({ onClickLoadMore }) => {
  return (
    <div className="w/3 flex m-auto justify-center py-4">
      <div
        className="text-violet-700 text-base font-semibold px-6 py-2 border-2 border-gray-500 rounded-lg cursor-pointer hover:bg-gray-200"
        onClick={onClickLoadMore}
      >
        Load More
      </div>
    </div>
  );
};
export default Pagination;
