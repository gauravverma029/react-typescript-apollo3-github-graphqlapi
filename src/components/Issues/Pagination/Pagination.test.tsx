import React from "react";
import { cleanup, render } from "@testing-library/react";

import Pagination from "./index";

afterEach(cleanup);
const handleLoadMore = () => {};

describe("Pagination Component", () => {
  test("Should Render", async () => {
    const { asFragment } = render(
      <Pagination onClickLoadMore={handleLoadMore} />
    );
    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Contain Text 'Load More'", async () => {
    const { getByText } = render(
      <Pagination onClickLoadMore={handleLoadMore} />
    );
    getByText("Load More");
  });

  test("Should Contain div tag for Load More text", async () => {
    const { getByText } = render(
      <Pagination onClickLoadMore={handleLoadMore} />
    );
    expect(getByText("Load More").tagName).toBe("DIV");
  });

  test("Should Contain ClassName", async () => {
    const { getByText } = render(
      <Pagination onClickLoadMore={handleLoadMore} />
    );
    expect(getByText("Load More").className).toBe(
      "text-violet-700 text-base font-semibold px-6 py-2 border-2 border-gray-500 rounded-lg cursor-pointer hover:bg-gray-200"
    );
  });
});
