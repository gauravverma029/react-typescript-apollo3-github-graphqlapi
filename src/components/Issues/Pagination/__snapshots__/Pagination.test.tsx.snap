// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Should Render 1`] = `
<DocumentFragment>
  <div
    class="w/3 flex m-auto justify-center py-4"
  >
    <div
      class="text-violet-700 text-base font-semibold px-6 py-2 border-2 border-gray-500 rounded-lg cursor-pointer hover:bg-gray-200"
    >
      Load More
    </div>
  </div>
</DocumentFragment>
`;
